### PR TITLE
Migration guide for Keycloak admin client

### LINKED ISSUE #20026: Migration guide for Keycloak admin client
### Description

In PR https://github.com/keycloak/keycloak/pull/19988, we've decided to change present artifacts as following:
`keycloak-admin-client` --> `keycloak-admin-client-jee` (JavaEE support)
`keycloak-admin-client-jakarta` --> `keycloak-admin-client` (JakartaEE support)

It should be mentioned in the migration guide, that these changes exists. 

For more information, take a look at the PR https://github.com/keycloak/keycloak/pull/19988.

### PR DESCRIPTION
**Pull request by mabartos**
Closes #20026 

Result for migration guide:
image

@vmuzikar @andymunro  Could you please check it? Thanks!

The Quarkus 3 release notes and migration guide will be prepared in a different PR.

### CODE REVIEW

File: docs/documentation/release_notes/topics/22_0_0.adoc
Comment (vmuzikar) [inline docs/documentation/release_notes/topics/22_0_0.adoc:19]:
```suggestion
After the upgrade to Jakarta EE, dependencies for Keycloak Admin clients were renamed to describe the reality more accurately, with consideration for long-term maintainability.
We still provide two separate Keycloak Admin clients, one with Jakarta EE and the other with Java EE support.
```
See e.g. https://jakarta.ee/, or [Wiki](https://en.wikipedia.org/wiki/Jakarta_EE).

File: docs/documentation/release_notes/topics/22_0_0.adoc
Comment (vmuzikar) [inline docs/documentation/release_notes/topics/22_0_0.adoc:18]:
```suggestion
After the upgrade to JakartaEE, artifacts for Keycloak Admin clients were renamed to describe the reality more accurately, with consideration for long-term maintainability.
```
Maybe we should call it "artifacts"? Not sure about that, but in any case I think dependency is not the right word for what we are distributing.

File: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
Comment (vmuzikar) [inline docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:132]: JakartaEE --> Jakarta EE
JavaEE --> Java EE
```suggestion
= Renamed Keycloak Admin client dependencies

After the upgrade to Jakarta EE, dependencies for Keycloak Admin clients were renamed to describe the reality more accurately, with consideration for long-term maintainability.
We still provide two separate Keycloak Admin clients, one with Jakarta EE and the other with Java EE support.

We stopped releasing the `org.keycloak:keycloak-admin-client-jakarta` dependency.
The default one for Keycloak Admin client with Jakarta EE support is `org.keycloak:keycloak-admin-client` (since version 22.0.0).

We will also release Keycloak Admin client with Java EE support for some time, but it's recommended to migrate to Jakarta EE as soon as possible.
The new dependency with Java EE support is `org.keycloak:keycloak-admin-client-jee`.

=== Jakarta EE support
[source,xml]
.Before migration:
----
<dependency>
    <groupId>org.keycloak</groupId>
    <artifactId>keycloak-admin-client-jakarta</artifactId>
    <version>21.0.0</version>
</dependency>
----

[source,xml]
.After migration:
----
<dependency>
    <groupId>org.keycloak</groupId>
    <artifactId>keycloak-admin-client</artifactId>
    <version>22.0.0</version>
</dependency>
----

=== Java EE support _(only temporary support)_
[source,xml]
.Before migration:
----
<dependency>
    <groupId>org.keycloak</groupId>
    <artifactId>keycloak-admin-client</artifactId>
    <version>21.0.0</version>
</dependency>
----

[source,xml]
.After migration:
----
<dependency>
    <groupId>org.keycloak</groupId>
    <artifactId>keycloak-admin-client-jee</artifactId>
    <version>22.0.0</version>
</dependency>
----
```

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: docs/documentation/release_notes/topics/22_0_0.adoc
Comment (mabartos) [reply docs/documentation/release_notes/topics/22_0_0.adoc]: :grin: Yep, agree. Like, in 5 minutes, I've changed the terminology like 5x when I was considering various points of view. But yeah, 'artifacts' might be better.

**Review by vmuzikar — APPROVED** [review]
@mabartos Thanks, LGTM.

Let's wait for a review from @andymunro to double check if the wording is good/polite enough. :)

File: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
Comment (vmuzikar) [inline docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:81]: Actually, one more. Sorry, missed it before.
```suggestion
= Renamed Keycloak Admin client artifacts
```

File: docs/documentation/release_notes/topics/22_0_0.adoc
Comment (vmuzikar) [inline docs/documentation/release_notes/topics/22_0_0.adoc:16]:
```suggestion
= Renamed Keycloak Admin client artifacts
```

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
Comment (mabartos) [reply docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:81]: Oh, me too. Thanks @vmuzikar!

**Review by vmuzikar — DISMISSED** [review]

File: docs/documentation/release_notes/topics/22_0_0.adoc
Comment (andymunro) [inline docs/documentation/release_notes/topics/22_0_0.adoc:16]:
```suggestion
= Renamed Keycloak Admin Console artifacts
```

File: docs/documentation/release_notes/topics/22_0_0.adoc
Comment (andymunro) [inline docs/documentation/release_notes/topics/22_0_0.adoc:19]:
```suggestion
After the upgrade to Jakarta EE, artifacts for Keycloak Admin Console were renamed to more descriptive names with consideration for long-term maintainability.
We still provide two separate Keycloak Admin Consoles, one with Jakarta EE and the other with Java EE support.
```

File: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
Comment (andymunro) [inline docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:77]:
```suggestion
In previous releases the `export` and `import` commands allowed runtime parameters such as a database URL only in configuration files or environment variables.
```

File: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
Comment (andymunro) [inline docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:81]:
```suggestion
= Renamed Keycloak Admin Console artifacts
```

File: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
Comment (andymunro) [inline docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:85]:
```suggestion
After the upgrade to Jakarta EE, artifacts for Keycloak Admin Console were renamed to more descriptive names with consideration for long-term maintainability.
We still provide two separate Keycloak Admin Consoles, one with Jakarta EE and the other with Java EE support.

```

File: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
Comment (andymunro) [inline docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:87]:
```suggestion
The default one for the Keycloak Admin Console with Jakarta EE support is `org.keycloak:keycloak-admin-client` (since version 22.0.0).
```

File: docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
Comment (andymunro) [inline docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:89]:
```suggestion
We will continue to release the Keycloak Admin Console with Java EE support for some time, but we recommend that you migrate to Jakarta EE as soon as possible.
```

**Review by andymunro — CHANGES_REQUESTED** [review]
@mabartos just a few changes.  I assume the Admin Client you mention is the Admin Console as described mainly in the Server Administration guide.  If not, please ignore those comments.

**Comment by mabartos** [comment]
@andymunro Thanks for your review! This is related to the Keycloak Admin client.

**Comment by mabartos** [comment]
@andymunro @vmuzikar I've addressed your suggestions. Thanks!

**Comment by mabartos** [comment]
@vmuzikar Could you please rerun the Documentation job in GHAs? Thanks

**Comment by vmuzikar** [comment]
@mabartos I re-ran the tests but I believe the failures were actually correct (invalid external links). But it's out of scope of this PR in any case.

@andymunro Thanks for the  feedback! Can you please approve if the changes look good to you when you have a chance?

**Review by andymunro — APPROVED** [review]
Looks good, @mabartos

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**
